src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -602,10 +609,32 @@ private async Task UploadFile(UploadFileInfo file)
                     // Create the log
                     var taskLog = await _jobServer.CreateLogAsync(_scopeIdentifier, _hubName, _planId, new TaskLog(String.Format(@"logs\{0:D}", file.TimelineRecordId)), default(CancellationToken));
 
-                    // Upload the contents
-                    using (FileStream fs = File.Open(file.Path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                    if (_writeToBlobstorageService)
                     {
-                        var logUploaded = await _jobServer.AppendLogContentAsync(_scopeIdentifier, _hubName, _planId, taskLog.Id, fs, default(CancellationToken));
+                        BlobIdentifierWithBlocks blobBlockId = null;
+                        using (FileStream fs = File.Open(file.Path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```

Comment:
nit: put this `using` before ` if (_writeToBlobstorageService)` to not duplicate it in both if and else